src/pkg/build/provisioners/provisioner_shub.go

```diff
@@ -9,23 +9,143 @@
 package provisioners
 
 import (
+	"encoding/json"
+        "errors"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+//        "log"
+	"os"
+        "time"
+        "strings"
 
+	"github.com/containers/image/docker"
+	"github.com/containers/image/types"
+	oci "github.com/containers/image/oci/layout"
 	"github.com/singularityware/singularity/src/pkg/image"
+	"github.com/singularityware/singularity/src/pkg/sylog"
 )
 
-// ShubProvisioner provisions a sandbox environment from a shub
-// URI.
+
+// ShubClient contains the uri and client
+type ShubClient struct {
+	Client   http.Client
+	ImageUri string
+	HTTPAddr string
+}
+
+// NewRemoteBuilder creates a RemoteBuilder with the specified details.
+func NewShubClient(uri string) (shubClient *ShubClient) {
+
+	sylog.Infof("%v\n", uri)
+	httpAddr := fmt.Sprintf("www.singularity-hub.org")
+	shubClient = &ShubClient{
+		Client: http.Client{
+			Timeout: 30 * time.Second,
+		},
+		HTTPAddr: httpAddr,
+		ImageUri: uri,
+	}
+
+	return
+}
+
+func (p *ShubProvisioner) getManifest() (err error) {
+
+	// Create a new Singularity Hub client
+	sc := &ShubClient{}
+
+        // TODO: need to parse the tag / digest and send along too
+        uri := strings.Split(p.srcRef.StringWithinTransport(), ":")[0]
+
+	// Format the http address, coinciding with the image uri
+	httpAddr := fmt.Sprintf("www.singularity-hub.org/api/container%s/", uri)
+	sylog.Infof("%v\n", httpAddr)
+
+	// Create the request, add headers context
+	url := url.URL{Scheme: "https", Host: sc.HTTPAddr, Path: httpAddr}
+        sylog.Infof("%v\n", url)
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+	if err != nil {
+		return
+	}
+
+	// Do the request, if status isn't success, return error
+	res, err := sc.Client.Do(req)
+	if err != nil {
+		return
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		err = errors.New(res.Status)
+		return
+	}
+
+	err = json.NewDecoder(res.Body).Decode(&p)
+	return
+
+}
+
+// NewShubProvisioner returns a provisioner that can dump a previously
+// build Singularity image into a sandbox based on a URI
+func NewShubProvisioner(src string) (p *ShubProvisioner, err error) {
+
+	// Question for bauerm - where does this get called?
```

Comment:
`NewShubProvisioner` gets called in `src/pkg/build/provisioner.go`, if that was the question :-)

Reply:
yes, thank you!